polygeist/lib/polygeist/Passes/ConvertPolygeistToLLVM.cpp

```diff
@@ -45,16 +49,513 @@
 using namespace mlir;
 using namespace polygeist;
 
-mlir::LLVM::LLVMFuncOp GetOrCreateMallocFunction(ModuleOp module);
-mlir::LLVM::LLVMFuncOp GetOrCreateFreeFunction(ModuleOp module);
+/// Conversion similar to the canonical one, but not inserting the obtained
+/// pointer in a struct.
+struct GetGlobalMemrefOpLowering
+    : public ConvertOpToLLVMPattern<memref::GetGlobalOp> {
+  using ConvertOpToLLVMPattern<memref::GetGlobalOp>::ConvertOpToLLVMPattern;
+
+  LogicalResult
+  matchAndRewrite(memref::GetGlobalOp getGlobalOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    const auto memrefTy = getGlobalOp.getType();
+    if (!canBeLoweredToBarePtr(memrefTy))
+      return failure();
+
+    const auto loc = getGlobalOp.getLoc();
```

Comment:
[nit]: There is only one use of `loc` at line 73 so you could use `getGlobalOp.getLoc()` directly and avoid creating the `loc` variable.